playbooks/nfs.yml

```diff
@@ -56,19 +65,31 @@
         mode: '755'
       loop: "{{ cifmw_nfs_shares }}"
 
-    - name: Set nfs network vars
+    - name: Set nfs network range
+      delegate_to: controller
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+        PATH: "{{ cifmw_path }}"
       vars:
         _nfs_network_name: "{{ cifmw_nfs_network | default('storage') }}"
-        _nfs_host: "{{ groups[cifmw_nfs_target | default('computes')][0] | default('') }}"
-      ansible.builtin.set_fact:
-        cifmw_nfs_network_range: >-
-          {{
-            crc_ci_bootstrap_networking.networks[_nfs_network_name].range
-          }}
-        cifmw_nfs_ip: >-
-          {{
-            crc_ci_bootstrap_networking.instances[_nfs_host].networks[_nfs_network_name].ip
-          }}
+        _nfs_host: "{{ [groups[cifmw_nfs_target | default('computes')][0], ansible_domain] | join('.') | default('') }}"
+        _ipset_namespace: "{{ cifmw_install_yamls_defaults['NAMESPACE'] | default('openstack') }}"
+      ansible.builtin.command:
+        cmd: oc get ipset {{ _nfs_host }} -n {{ _ipset_namespace }} -o jsonpath='{.status.reservations[?(@.network=="{{ _nfs_network_name }}")].cidr}'
+      register: cifmw_nfs_network_range_out
+
+    - name: Set nfs ip
+      delegate_to: controller
+      environment:
+        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+        PATH: "{{ cifmw_path }}"
+      vars:
+        _nfs_network_name: "{{ cifmw_nfs_network | default('storage') }}"
+        _nfs_host: "{{ [groups[cifmw_nfs_target | default('computes')][0], ansible_domain] | join('.') | default('') }}"
+        _ipset_namespace: "{{ cifmw_install_yamls_defaults['NAMESPACE'] | default('openstack') }}"
+      ansible.builtin.command:
+        cmd: oc get ipset {{ _nfs_host }} -n {{ _ipset_namespace }} -o jsonpath='{.status.reservations[?(@.network=="{{ _nfs_network_name }}")].address}'
+      register: cifmw_nfs_ip_out
```

Comment:
I assume you're suggesting to make mutiple api calls like now but in a loop. I would prefer to keep it multiple tasks rather than using indexes from results to avoid complexity. This is part of a hook only used when using nfs backend for glance and could be optimized, if required later.